environment.py

```diff
@@ -84,29 +84,42 @@ def findCellNeighbors(self):
 
     def findCellsInCardinalRange(self, startX, startY, gridRange):
-        cellsInRange = []
+        cellsInRange = [{"cell": self.grid[startX][startY], "distance": 0}]
+        height = self.height
+        width = self.width
         for i in range(1, gridRange + 1):
-            deltaNorth = (startY + i + self.height) % self.height
-            deltaSouth = (startY - i + self.height) % self.height
-            deltaEast = (startX + i + self.width) % self.width
-            deltaWest = (startX - i + self.width) % self.width
-            cellsInRange.append({"cell": self.grid[startX][deltaNorth], "distance": i})
-            cellsInRange.append({"cell": self.grid[startX][deltaSouth], "distance": i})
-            cellsInRange.append({"cell": self.grid[deltaEast][startY], "distance": i})
-            cellsInRange.append({"cell": self.grid[deltaWest][startY], "distance": i})
+            deltaNorth = (startY + i + height) % height
+            deltaSouth = (startY - i + height) % height
+            deltaEast = (startX + i + width) % width
+            deltaWest = (startX - i + width) % width
+            cellsInRange.extend([
+                {"cell": self.grid[startX][deltaNorth], "distance": i},
+                {"cell": self.grid[startX][deltaSouth], "distance": i},
+                {"cell": self.grid[deltaEast][startY], "distance": i},
+                {"cell": self.grid[deltaWest][startY], "distance": i}])
         return cellsInRange
 
     def findCellsInRadialRange(self, startX, startY, gridRange):
-        cellsInRange = []
-        # Iterate through the bounding box of the circle
-        for i in range(startX - gridRange, startX + gridRange + 1):
-            for j in range(startY - gridRange, startY + gridRange + 1):
+        cellsInRange = self.findCellsInCardinalRange(startX, startY, gridRange)
+        height = self.height
+        width = self.width
+        # Iterate through the upper left quadrant of the circle's bounding box
+        for i in range(startX - gridRange, startX):
+            for j in range(startY - gridRange, startY):
                 euclideanDistance = math.sqrt(pow((i - startX), 2) + pow((j - startY), 2))
                 # If agent can see at least part of a cell, they should be allowed to consider it
                 if euclideanDistance < gridRange + 1:
-                    deltaX = (i + self.height) % self.height
-                    deltaY = (j + self.width) % self.width
-                    cellsInRange.append({"cell": self.grid[deltaX][deltaY], "distance": euclideanDistance})
+                    # Coordinates for the top left quadrant
+                    deltaX = (i + height) % height
+                    deltaY = (j + width) % width
+                    # Reflected coordinates for the other three quadrants
+                    reflectedX = (2 * startX - i + height) % height
+                    reflectedY = (2 * startY - j + width) % width
+                    cellsInRange.extend([
```

Comment:
Same as above, compress if possible.

Reply:
Compressed in new commit, but I'm not sure if it's the style you're looking for. Let me know.

Reply:
I think it reads better to have it expanded rather than in a list comprehension (so how you had it before) just not one on each line. See if it looks nice with 2 per line or all on 1 line.

Reply:
1 line looks too long to me. 2 is alright for the variables, but then you have to have 4 lines for the `extend()`.